pkg/router/template/router.go

```diff
@@ -450,11 +458,38 @@ func (r *templateRouter) AddRoute(id string, route *routeapi.Route, host string)
 		}
 	}
 
+	var err error
```

Comment:
(like unique hosts and the status update).  I ask because this structure makes it harder to refactor the router to ingress - while the admission approach means that by the time the template router gets the code it's known "good".